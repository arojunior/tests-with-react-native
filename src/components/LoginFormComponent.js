import React from 'react';
import PropTypes from 'prop-types';
import { Content, Form, Item, Input, Label, Button, Text } from 'native-base';

const LoginFormComponent = ({ handleSubmit, handleChange, errors }) => (
  <Form>
    <Item floatingLabel error={!!errors.username}>
      <Label>Username</Label>
      <Input onChangeText={handleChange(`username`)} testID="username_input" />
    </Item>
    {errors.username && (
      <Item>
        <Text>{errors.username}</Text>
      </Item>
    )}
    <Item floatingLabel last error={!!errors.password}>
      <Label>Password</Label>
      <Input
        secureTextEntry
        onChangeText={handleChange(`password`)}
        testID="password_input"
      />
    </Item>
    {errors.password && (
      <Item>
        <Text>{errors.password}</Text>
      </Item>
    )}
    <Content padder>
      <Button
        block
        onPress={handleSubmit}
        testID="login_button"
      >
        <Text>Login</Text>
      </Button>
    </Content>
  </Form>
);

LoginFormComponent.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default LoginFormComponent;
