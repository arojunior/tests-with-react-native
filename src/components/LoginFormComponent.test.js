import LoginFormComponent from './LoginFormComponent';

describe(`LoginFormComponent`, () => {
  test(`Snapshot`, () => {
    expect(LoginFormComponent).toMatchSnapshot();
  });
});
