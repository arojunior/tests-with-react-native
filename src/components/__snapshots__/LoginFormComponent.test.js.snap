// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormComponent Snapshot 1`] = `[Function]`;
